dbt/adapters/databricks/impl.py

```diff
@@ -175,6 +158,19 @@ class DatabricksAdapter(SparkAdapter):
         }
     )
 
+    get_column_behavior: GetColumnsBehavior
+
+    def __init__(self, config: Any, mp_context: SpawnContext) -> None:
```

Comment:
Although I don't control the invocation of init, and thus can't pass the behaviors in, doing this during init ensures we only check the behavior flag once.  I tried to make this a little more functional but couldn't figure out how to override an existing function definition that is inherited from a parent, hence the goofy class-based strategy.